packages/core/src/percy.js

```diff
@@ -213,55 +217,62 @@ export class Percy {
   async *stop(force) {
     // not started, but the browser was launched
-    if (!this.readyState && this.browser.isConnected()) {
-      await this.browser.close();
-    }
-
-    if (this.syncQueue) this.syncQueue.stop();
-    // not started or already stopped
-    if (!this.readyState || this.readyState > 2) return;
-
-    // close queues asap
-    if (force) {
-      this.#discovery.close(true);
-      this.#snapshots.close(true);
-    }
-
-    // already stopping
-    if (this.readyState === 2) return;
-    this.readyState = 2;
-
-    // log when force stopping
-    if (force) this.log.info('Stopping percy...');
-
-    // used to log snapshot count information
-    let info = (state, size) => `${state} ` +
-      `${size} snapshot${size !== 1 ? 's' : ''}`;
-
     try {
-      // flush discovery and snapshot queues
-      yield* this.yield.flush((state, size) => {
-        this.log.progress(`${info(state, size)}...`, !!size);
-      });
-    } catch (error) {
-      // reset ready state when aborted
-      /* istanbul ignore else: all errors bubble */
-      if (error.name === 'AbortError') this.readyState = 1;
-      throw error;
-    }
-
-    // if dry-running, log the total number of snapshots
-    if (this.dryRun && this.#snapshots.size) {
-      this.log.info(info('Found', this.#snapshots.size));
+      if (!this.readyState && this.browser.isConnected()) {
+        await this.browser.close();
+      }
+  
+      if (this.syncQueue) this.syncQueue.stop();
+      // not started or already stopped
+      if (!this.readyState || this.readyState > 2) return;
+  
+      // close queues asap
+      if (force) {
+        this.#discovery.close(true);
+        this.#snapshots.close(true);
+      }
+  
+      // already stopping
+      if (this.readyState === 2) return;
+      this.readyState = 2;
+  
+      // log when force stopping
+      if (force) this.log.info('Stopping percy...');
+  
+      // used to log snapshot count information
+      let info = (state, size) => `${state} ` +
+        `${size} snapshot${size !== 1 ? 's' : ''}`;
+  
+      try {
+        // flush discovery and snapshot queues
+        yield* this.yield.flush((state, size) => {
+          this.log.progress(`${info(state, size)}...`, !!size);
+        });
+      } catch (error) {
+        // reset ready state when aborted
+        /* istanbul ignore else: all errors bubble */
+        if (error.name === 'AbortError') this.readyState = 1;
+        throw error;
+      }
+  
+      // if dry-running, log the total number of snapshots
+      if (this.dryRun && this.#snapshots.size) {
+        this.log.info(info('Found', this.#snapshots.size));
+      }
+  
+      // close server and end queues
+      await this.server?.close();
+      await this.#discovery.end();
+      await this.#snapshots.end();
+  
+      // mark instance as stopped
+      this.readyState = 3;
+    } catch(err) {
+      this.log.error(err)
```

Comment:
In discovery we are only collecting error. Shouldn't we capture the time taken and all for the same?

Reply:
Adding timing for asset discovery using performance api